kyaml/filesys/filesystem_test.go

```diff
@@ -42,3 +52,87 @@ func testNotExistErr(t *testing.T, fs FileSystem) {
 	err = fs.Walk(path, func(_ string, _ os.FileInfo, err error) error { return err })
 	assert.Truef(t, errors.Is(err, os.ErrNotExist), "Walk should return ErrNotExist, got %v", err)
 }
+
+// setupFileSys returns file system, default directory to test in, and working directory
+func setupFileSys(t *testing.T, name string) (FileSystem, string, string) {
+	t.Helper()
+
+	switch name {
+	case disk:
+		fSys, testDir := makeTestDir(t)
+		return fSys, testDir, cleanWd(t)
+	case memoryAbs:
+		return filesysBuilders[name](), Separator, Separator
+	default:
+		require.Equal(t, memoryEmpty, name)
+		return filesysBuilders[name](), "", ""
+	}
+}
+
+func TestDemandDir(t *testing.T) {
+	for name := range filesysBuilders {
+		thisName := name
```

Comment:
Any reason to not just do `name := name` here? Why change it to `thisName`?

Reply:
Not really. I just thought maybe `thisName` would convey that I created a new variable to hold onto the value of _this_ `name` lol. I guess it didn't come through?

Reply:
Ah. Up to you in that case, I'll defer to the style that you prefer.